Tests/E2E/LoRaAPIHelper.cs

```diff
@@ -43,6 +44,15 @@ public static async Task<bool> ResetADRCache(string devEUI)
             return response.IsSuccessStatusCode;
         }
 
+        public static async Task<bool> SendCloudToDeviceMessage(string devEUI, LoRaCloudToDeviceMessage c2dMessage)
+        {
+            var url = new Uri($"{baseUrl}cloudtodevicemessage/{devEUI}?code={authCode}");
```

Comment:
[nit] we could extract the ceremony of the url creation, `authCode` appending and posting etc in a method. I had something of the following signature in mind `static async Task<bool> PostFunctionEndpointAsync(string path, HttpContent content, CancellationToken cancellationToken)`